lib/src/struct_annotation.dart

```diff
@@ -227,14 +228,43 @@ macro class Struct with _Shared implements ClassDeclarationsMacro, ClassDefiniti
           clazzName,
           '(',
           for (final field in fields)
-          ...[field.identifier.name,': ', field.identifier.name, ' ?? ', 'this.',field.identifier.name, ','],
+            ...[field.identifier.name,': ', field.identifier.name, ' ?? ', 'this.',field.identifier.name, ','],
           ');'
         ],
       ),
     );
   }
+
+  Future<void> _declareDefaultConstructor(
+      ClassDeclaration clazz,
+      MemberDeclarationBuilder builder,
+      ) async {
+    final fieldDeclarations = await builder.fieldsOf(clazz);
+    final fields = await Future.wait(
+      fieldDeclarations.map(
+            (f) async => (
+        identifier: f.identifier,
+        type: _checkNamedType(f.type, builder),
+        ),
+      ),
+    );
+
+    final missingType = fields.firstWhereOrNull((f) => f.type == null);
+    if (missingType != null) return null;
+
+    return builder.declareInType(
+      DeclarationCode.fromParts([
+        '${clazz.identifier.name}({',
+        for (final field in fields)
+          ...['required', ' ', 'this.', field.identifier.name, ','],
+        '});',
+      ]),
+    );
+  }
+
 }
 
+
```

Comment:
nit can we remove the additional newline here?

Reply:
See https://github.com/felangel/struct/pull/8/commits/bfa367f7a609303bf0bd844427fa0295e8995229